### ISSUE TITLE
Allow offline bootstrapping

### ISSUE THREAD

**Issue by reubenmiller** [thin-edge/thin-edge.io#2831]
**Is your feature improvement request related to a problem? Please describe.**

Bootstrapping a device is not possible when the device does not have an internet connection, as it causes the `tedge connect c8y` to fail, which then deletes the mosquitto bridge configuration.

Offline bootstrapping is useful in the factory scenario, where the factory devices should be on-boarded, however they do not have an internet connection to Cumulocity IoT (due to security constraints of the factory network). In this scenario the bootstrapping would require a separate machine (which has internet connection) to facility that pre-registration of the device in Cumulocity IoT, and create/sign the appropriate device certificates and transfer them to the device.

**Describe the solution you'd like**

Add an option to generate the required MQTT bridge connection without doing the network test, so that the thin-edge.io instance can be left in a state where it will connect as soon as device is able to reach Cumulocity IoT (e.g. installed in a network with internet connectivity):

For example, a new `--offline` flag could be added to the `tedge connect <cloud>` command to ignore connection registration.

```sh
tedge connect c8y --offline
```

**Note** In order for the above to work, the device must be pre-registered in Cumulocity IoT via the [bulk registration api](https://cumulocity.com/api/core/2024/#operation/postBulkNewDeviceRequestCollectionResource) (see the **Additional context** section for background.)

**Describe alternatives you've considered**

**Additional context**

In Cumulocity IoT, a device can be preregistered using the [bulk registration API](https://cumulocity.com/api/core/2024/#operation/postBulkNewDeviceRequestCollectionResource). This step could be added to the `c8y-tedge` extension which could be performed when running `c8y tedge bootstrap root@mydevice.local`.

Below shows an example (using go-c8y-cli) where a given device is pre-registered which will create a device managed object in the platform:

```sh
COMMON_NAME=gateway00001
DEVICE_TYPE=thin-edge.io

c8y api POST /devicecontrol/bulkNewDeviceRequests --file <(cat << EOT
ID;AUTH_TYPE;TYPE;NAME;com_cumulocity_model_Agent.active
"$COMMON_NAME";CERTIFICATES;"${DEVICE_TYPE}";"$COMMON_NAME";true
EOT
```

**Comment by Ruadhri17** [comment]
Resolved by #2843

Ticket introduces new `--offline` flag to `tedge connect` command that allows to create bridge configuration when there is no access to internet connection. When internet connection is restored, device will connect to the cloud. Work with all supported clouds: `c8y`, `aws` and `az`.  

Following doc sections were updated [c8y](https://github.com/thin-edge/thin-edge.io/blob/main/docs/src/start/connect-c8y.md), [azure](https://github.com/thin-edge/thin-edge.io/blob/main/docs/src/start/connect-azure.md), [aws](https://github.com/thin-edge/thin-edge.io/blob/main/docs/src/start/connect-aws.md) and [tedge connect cli](https://github.com/thin-edge/thin-edge.io/blob/main/docs/src/references/cli/tedge-connect.md).

System test exists and it can be found [here](https://github.com/thin-edge/thin-edge.io/blob/main/tests/RobotFramework/tests/tedge_connect/tedge_connect_offline.robot).

**Comment by gligorisaev** [comment]
QA has thoroughly checked the feature and here are the results:

- Checked the functioning according the documentation and it works fine
- Documentation changes have been reviewed and the description is accurate
- Test validating the feature is created and reviewed:
   - tests/RobotFramework/tests/tedge_connect/tedge_connect_offline.robot
- QA has tested the function and it's functioning according description.
